modules/likes/jetpack-likes-settings.php

```diff
@@ -590,22 +590,22 @@ function is_comments_enabled() {
 	function admin_settings_callback() {
 		// We're looking for these, and doing a dance to set some stats and save
 		// them together in array option.
-		$new_state = ! empty( $_POST['wpl_default'] ) ? $_POST['wpl_default'] : 'on';
+		$new_state = ! empty( $_POST['wpl_default'] ) ? $_POST['wpl_default'] : false;
 		$db_state  = $this->is_enabled_sitewide();
 
-		$reblogs_new_state = ! empty( $_POST['jetpack_reblogs_enabled'] ) ? $_POST['jetpack_reblogs_enabled'] : 'on';
+		$reblogs_new_state = ! empty( $_POST['jetpack_reblogs_enabled'] ) ? $_POST['jetpack_reblogs_enabled'] : false;
```

Comment:
I'm sorry, I don't understand why you are changing the default value from positive to negative? Should this be `true` instead?

Reply:
Yes, this was kind of confusing to me ... I will try to explain. Since I changed the input values to `"0"` and `"1"` on lines 497 and 503, when the option is disabled, [`empty("0")` evaluates to `true`](https://www.php.net/manual/en/function.empty.php) and thus uses the `false` / `else` side of the ternary -- this is no longer a default value for when no input value is passed in, but rather the value when "0" is passed in (disabled). When the values were `on` and `off`, those strings were never considered `empty` and so the `else` side of the ternary was actually a default value.

If I put `true` there, it is impossible to disable the option, it always appears as enabled on page load.

Reply:
Note that on line 635 there is similar behavior (I think??), where the default value for the ternary is `false`... I assume 0 / 1 are being passed in there.